### ISSUE TITLE
Can't add Swift file to Objetive-C pod library

### ISSUE THREAD

**Issue by ItayAmzaleg** [CocoaPods/CocoaPods#7404]
* [ ] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

I'm trying to add a swift file to a working obj-c pod project.
I'm not really sure if it's even possible.
I've added the file to the project, Got no prompt for creating bridging header, I guess I don't need one for a library.
On podspec I set swift version `s.swift_version = '4.0'` and source files to all files in all folders recursively `s.source_files = 'MyPod/Classes/**/*'`.

## What did you expect to happen?

Have an hybrid Pod project. or at least get compilation error about swift file or missing declarations.

## What happened instead?

Cant validate podspec file.
While `pod spec lint --allow-warnings --no-clean` I see the swift file in my Pod library, but the lint is failing on
`- ERROR | [iOS] xcodebuild:  /Users/my.user/Library/Developer/Xcode/DerivedData/App-compwrxznydejjfbpuppnmocqadi/Build/Products/Release-iphonesimulator/MyPod/MyPod.framework/Headers/MHJsonRequestSerializer.h:9:9: error: 'AFURLRequestSerialization.h' file not found
    - NOTE  | [iOS] xcodebuild:  <unknown>:0: error: could not build Objective-C module 'MyPod'`

### Stack

```
   CocoaPods : 1.4.0
        Ruby : ruby 2.3.3p222 (2016-11-21 revision 56859) [universal.x86_64-darwin17]
    RubyGems : 2.5.2
        Host : Mac OS X 10.13.3 (17D47)
       Xcode : 9.2 (9C40b)
         Git : git version 2.15.1
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 9593bf3a0c89db304c44bb3acb8ba35b02eb8eae
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.2
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.0
cocoapods-try         : 1.1.0
```

**Comment by dnkoutso** [comment]
You are using the `--no-clean` option so open the generated workspace and inspect it yourself.

At this time this most likely seems to be a user or pod configuration issue and not a cocoapods library issue. It is best to ask these questions on Stackoverflow under `cocoapods` tag. 

If you believe this is a cocoapods library issue then please upload a sample app demonstrating the issue exactly and I will re-open.

**Comment by ItayAmzaleg** [comment]
I did ask the same question on [stackoverflow](https://stackoverflow.com/questions/48607024/add-swift-to-existing-cocoapods-objective-c-library).
Obviously I've opened the generated workspace, while compiling I get the same error `file not found`. When I drill down I can see the file is there. when removing the added swift file workspace compile just fine. 
I don't know if that's related, but the header file cannot be found is an AFNetwork header (which is a dependency).
I can't find example for a hybrid podspec configuration, so I don't know if it's a configuration issue. I posted here because my pod works fine without swift and can't compile with swift.
